cavern/src/test/java/org/opencadc/cavern/files/CavernURLGeneratorTest.java

```diff
@@ -249,158 +249,174 @@ public void testRoundTripSuccess() {
+//            Assert.fail("unexpected exception: " + unexpected);
+//        }
+//    }
+
+//    // TODO: acl specific codes will be moved to a library, enable the test after
+//    @Ignore
+//    @Test
+//    public void testMetaTampered() {
+//        // TODO: test stil relevant?
+//        try {
+//
+//            TestTransferGenerator urlGen = new TestTransferGenerator(ROOT);
+//            VOSURI nodeURI = new VOSURI(baseURI + "/" + TEST_DIR + "/" + TEST_FILE);
+//            List<Protocol> protos = new ArrayList<>();
+//            protos.add(new Protocol(VOS.PROTOCOL_HTTPS_GET));
+//            final Transfer trans = new Transfer(nodeURI.getURI(), Direction.pullFromVoSpace);
+//            trans.getProtocols().addAll(protos);
+//            View view = null;
+//            Job job = null;
+//            List<Protocol> result = urlGen.getEndpoints(nodeURI, trans, view, job, null);
+//            Protocol p = result.get(0);
+//            Assert.assertNotNull(p);
+//            String suri = p.getEndpoint();
+//            log.debug("Transfer URI: " + suri);
+//            Assert.assertNotNull(suri);
+//            URI transferURI = new URI(suri);
+//
+//            Assert.assertTrue(transferURI.getPath().endsWith("/" + TEST_FILE));
+//
+//            String path = transferURI.getPath();
+//            log.debug("Path: " + path);
+//            // Have to generate a token with different meta and pass it in, using token Tool?
+////            String[] parts = path.split("/");
+////            URI targetURI = urlGen.getURIFromPath(path);
+//////            String sig = parts[4];
+////            //String meta = parts[3];
+////            String token = parts[3];
+////            VOSURI altURI = new VOSURI(baseURI + "/" + TEST_DIR + "/fakeFile");
+//////            String meta = new String(Base64.encode(("node=" + altURI.toString() + "&dir=pullFromVoSpace").getBytes()));
+////            try {
+//////                VOSURI retURI = urlGen.getNodeURI(meta, sig, Direction.pullFromVoSpace);
+////                VOSURI retURI = urlGen.getNodeURI(token, targetURI, Direction.pullFromVoSpace);
+////                Assert.fail();
+////            } catch (AccessControlException e) {
+////                // expected
+////            }
+//
+//        } catch (Exception unexpected) {
+//            log.error("unexpected exception", unexpected);
+//            Assert.fail("unexpected exception: " + unexpected);
+//        }
+//    }
+
+//    @Test
+//    public void testBase64URI() {
+//        // not relelvant as base64URLDecode not in CavernURLGenerator anymore
+//        String[] testStrings = {
+//            "abcde",
+//            "ab//de",
+//            "ab/de",
+//            "ab++///+de"
+//        };
+//        for (String s : testStrings) {
+//            log.debug("testing: " + s);
+//            Assert.assertEquals(CavernURLGenerator.base64URLDecode(CavernURLGenerator.base64URLEncode(s)), s);
+//        }
+//    }
```

Comment:
Is there a reason all these lines have been commented out?

Reply:
In the end, the tests that were commented out were either items tested in the TokenTool suite (as the code is entirely there,) and/or were using the old meta & sig format instead of token.  Updated the tests to exercise code in validateToken (was 'getNodeURI') re: testing direction against Grant class, and base 64 decode of the token before passing into token tool (as that's an extra step performed in CavernURLGenerator where the preauth URL with token is created.)